### PR TITLE
Nested `null` values do not remove keys from sub-charts

### LINKED ISSUE #9136: Nested `null` values do not remove keys from sub-charts
#2648 and #5184 allow for deletion of keys by setting null values in a values.yaml file. However it doesn't work for removing nested values in sub-charts, e.g.:

values.yaml:

```yaml
kibana:
  resources:
    requests:
      cpu: 1000m
      memory: 2Gi
    limits:
      cpu: null
      memory: 2Gi
```

Chart.yaml:
```yaml
apiVersion: v2
name: elastic
description: Elastic stack
type: application
version: 0.0.1
dependencies:
  - name: kibana
    version: 7.9.3
    repository: "@elastic"
```

Results in a deployment object with the base cpu limits:
```yaml
...
resources:
  limits:
    cpu: 1000m
    memory: 2Gi
  requests:
    cpu: 1000m
    memory: 2Gi

```

Output of `helm version`:

```bash
version.BuildInfo{Version:"v3.4.1", GitCommit:"c4e74854886b2efe3321e185578e6db9be0a6e29", GitTreeState:"clean", GoVersion:"go1.14.11"}
```

Output of `kubectl version`:

```bash
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"archive", BuildDate:"2020-12-04T15:20:33Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.10", GitCommit:"62876fc6d93e891aa7fbe19771e6a6c03773b0f7", GitTreeState:"clean", BuildDate:"2020-10-15T01:43:56Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

AWS/Kops

### PR DESCRIPTION
**Pull request by elliotkennedy**
**What this PR does / why we need it**:

closes #9136 

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility - **Need to verify import values are not broken**

### CODE REVIEW

**Comment by alexrashed** [comment]
Is there anything I could do to help getting this PR ready to be merged?

I'm facing this problem among several charts (for[ codecentric's keycloak](https://artifacthub.io/packages/helm/codecentric/keycloak), [codecentric's mailhog](https://artifacthub.io/packages/helm/codecentric/mailhog), and [elastic's elasticsearch](https://artifacthub.io/packages/helm/elastic/elasticsearch) as subcharts) in an OpenShift environment (where I want to set the `securityContext` to `null` in order to use the default by OpenShift).

File: pkg/chartutil/dependencies.go [hunk 1]
```diff
@@ -19,15 +19,34 @@ import (
...
 
+	"github.com/mitchellh/copystructure"
+
 	"helm.sh/helm/v3/pkg/chart"
 )
 
 // ProcessDependencies checks through this chart's dependencies, processing accordingly.
-func ProcessDependencies(c *chart.Chart, v Values) error {
-	if err := processDependencyEnabled(c, v, ""); err != nil {
+func ProcessDependencies(c *chart.Chart, vals map[string]interface{}) error {
```
Comment (bacongobbler) [inline pkg/chartutil/dependencies.go:28]: Given that this is a public exported function, we cannot change the function signature until Helm 4. Is there a way to rewrite this PR without requiring the need to rewrite the signature?

An alternative would be to introduce a new function and document `ProcessDependencies` as marked for deprecation in Helm 4.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by bacongobbler** [comment]
ping.

**Comment by elliotkennedy** [comment]
> ping.

Hey @bacongobbler - thanks for looking at this. Picking up changes now.

**Comment by Gollam** [comment]
@bacongobbler what's the status of this?

File: pkg/chartutil/dependencies.go (hunk 1 above)
Comment (elliotkennedy) [reply pkg/chartutil/dependencies.go]: I've deprecated the ProcessDependencies method for ProcessDependenciesWithValues

**Comment by elliotkennedy** [comment]
@bacongobbler could you please re-review? Thanks

**Comment by pisto** [comment]
Waiting for this fix.

**Comment by chattarajoy** [comment]
Waiting for this fix

**Comment by Imunhatep** [comment]
PR is here, and this bug leaves for years!!! unbelievable...
@bacongobbler [bacongobbler](https://github.com/bacongobbler)

**Comment by TheRealDwidwi** [comment]
any updates about this ?

**Comment by sabre1041** [comment]
@elliotkennedy I have gone through this PR and have been unsuccessful attempting to both integrate it into the current Helm codebase as a number of changes have occurred since this PR was submitted. Tests also do not currently pass either based on my initial attempts.

Would you be able to rebase this PR against `main` and see if you might be able to address the current test failures so that we can look towards integrating this feature?

**Comment by jamoflaw** [comment]
@sabre1041 FYI I believe https://github.com/helm/helm/pull/11440 should resolve this as an alternative to this PR

**Comment by sabre1041** [comment]
> @sabre1041 FYI I believe #11440 should resolve this as an alternative to this PR

I'll take a look!

**Comment by mattfarina** [comment]
I believe https://github.com/helm/helm/pull/12162 fixed this issue. If it's not fixed, please let us know.

**Pull request closed without merging**